uaclient/actions.py

```diff
@@ -51,6 +57,26 @@ def attach_with_token(
     update_apt_and_motd_messages(cfg)
 
 
+def magic_attach(cfg: config.UAConfig, magic_id: str, num_min_to_wait: int):
+    ua_contract = contract.UAContractClient(cfg=cfg)
+
+    while num_min_to_wait > 0:
+        try:
+            response = ua_contract.request_magic_token(magic_id=magic_id)
+            return MagicAttachResult(
+                contract_token=response.get("contract_token"),
+            )
+        except exceptions.MagicAttachIDNotResolved:
+            num_min_to_wait -= 1
+            time.sleep(60)
```

Comment:
I think the timeout should probably be specified in seconds, and the time between polls should be much lower, like 5 seconds. We don't want a user to have to wait a full minute after clicking the button in their browser for their install to continue.

Reply:
If someone knows what they're doing and is prepared (already has browser open, logged in, knows what subscription they're going to select or will use auto-selected free subscription), I could see it only taking a couple seconds.

I don't think there's a downside to more frequent polling aside from extra requests/traffic to the server, but I think CS is expecting more frequent polling - @alnvdl how often were you expecting `ua` to poll for the magic-attach?